tests/prague/eip7623_increase_calldata_cost/test_refunds.py

```diff
@@ -0,0 +1,289 @@
+    execution_gas_used: int,
+    max_refund: int,
+) -> int:
+    """Return the refund gas of the transaction."""
+    total_gas_used = tx_intrinsic_gas_cost_before_execution + execution_gas_used
+    return min(max_refund, total_gas_used // 5)
+
+
+@pytest.fixture
+def to(
+    pre: Alloc,
+    execution_gas_used: int,
+    prefix_code: Bytecode,
+    prefix_code_gas: int,
+    code_storage: Dict,
+) -> Address | None:
+    """
+    Return a contract that consumes the expected execution gas.
+
+    At the moment we naively use JUMPDEST to consume the gas, which can yield very big contracts.
+
+    Ideally, we can use memory expansion to consume gas.
+    """
+    extra_gas = execution_gas_used - prefix_code_gas
+    return pre.deploy_contract(
+        prefix_code + (Op.JUMPDEST * extra_gas) + Op.STOP,
+        storage=code_storage,
+    )
+
+
+@pytest.fixture
+def tx_gas_limit(
+    tx_intrinsic_gas_cost_including_floor_data_cost: int,
+    tx_intrinsic_gas_cost_before_execution: int,
+    execution_gas_used: int,
+) -> int:
+    """
+    Gas limit for the transaction.
+
+    The gas delta is added to the intrinsic gas cost to generate different test scenarios.
+    """
+    tx_gas_limit = tx_intrinsic_gas_cost_before_execution + execution_gas_used
+    assert tx_gas_limit >= tx_intrinsic_gas_cost_including_floor_data_cost
+    return tx_gas_limit
+
+
+@pytest.mark.parametrize(
+    "refund_test_type",
+    [
+        RefundTestType.EXECUTION_GAS_MINUS_REFUND_GREATER_THAN_DATA_FLOOR,
+        RefundTestType.EXECUTION_GAS_MINUS_REFUND_LESS_THAN_DATA_FLOOR,
+        RefundTestType.EXECUTION_GAS_MINUS_REFUND_EQUAL_TO_DATA_FLOOR,
+    ],
+)
+@pytest.mark.parametrize(
+    "refund_type",
+    [
+        RefundType.STORAGE_CLEAR,
+        RefundType.STORAGE_CLEAR | RefundType.AUTHORIZATION_EXISTING_AUTHORITY,
+        RefundType.AUTHORIZATION_EXISTING_AUTHORITY,
+    ],
+)
+def test_gas_refunds_from_data_floor(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    tx: Transaction,
+    tx_floor_data_cost: int,
+    tx_intrinsic_gas_cost_before_execution: int,
+    execution_gas_used: int,
+    refund: int,
+    refund_test_type: RefundTestType,
+) -> None:
+    """Test gas refunds deducted from the execution gas cost and not the data floor."""
+    gas_used = tx_intrinsic_gas_cost_before_execution + execution_gas_used - refund
+    if refund_test_type == RefundTestType.EXECUTION_GAS_MINUS_REFUND_LESS_THAN_DATA_FLOOR:
+        assert gas_used < tx_floor_data_cost
+    elif refund_test_type == RefundTestType.EXECUTION_GAS_MINUS_REFUND_GREATER_THAN_DATA_FLOOR:
+        assert gas_used > tx_floor_data_cost
+    elif refund_test_type == RefundTestType.EXECUTION_GAS_MINUS_REFUND_EQUAL_TO_DATA_FLOOR:
+        assert gas_used == tx_floor_data_cost
+    else:
+        raise ValueError("Invalid refund test type")
+    if gas_used < tx_floor_data_cost:
+        gas_used = tx_floor_data_cost
+    tx.expected_receipt = TransactionReceipt(gas_used=gas_used)
```

Comment:
so all the actual verification happens here. that gas_used we calculated is as t8n returns? need to hint it in a comment.

Reply:
Added a comment, but I'm thinking that, to make it easier when debugging the test, we could include the `gas_used` field in the state test fixture.

Potentially here:
https://github.com/ethereum/execution-spec-tests/blob/4891ac729e654ec361d2430960a9b307be125f9a/src/ethereum_test_fixtures/state.py#L86-L94

Reply:
Created an issue #1115